### PR TITLE
Config.cmake.in: Set project_LIBRARIES - fix issue #346

### LINKED ISSUE #346: project_LIBRARIES is not exported in CMake config
This is the source of the build/release regression in EigenPy between v1.6.x and v2.x and analog to #343: LIBRARIES does not get set, i.e., projects relying on it (e.g. MoveIt) are broken.

### PR DESCRIPTION
**Pull request by wxmerkt**
Resolves #346

### CODE REVIEW

**Comment by wxmerkt** [comment]
This is not yet fully working due to CMake requiring full paths to libraries. This was originally fixed in https://github.com/stack-of-tasks/eigenpy/blob/65ade259193c85c79701530bae70f80bc3af05d8/eigenpyConfig.cmake

**Comment by wxmerkt** [comment]
@jcarpent Can you please have a look at this and let me know if there is a better solution? This 
fixes the regression with MoveIt from EigenPy

File: Config.cmake.in [hunk 1]
```diff
@@ -7,6 +7,18 @@ set("@PROJECT_NAME_UPPER@_DOXYGENDOCDIR" "${PACKAGE_PREFIX_DIR}/share/doc/@PROJE
...
 set("@PROJECT_NAME@_PKG_CONFIG_DEPENDENCIES" "@_PKG_CONFIG_REQUIRES_LIST@")
 
+# Find absolute library paths for all _PKG_CONFIG_LIBS as CMake expects full paths, while pkg-config does not.
+set(@PROJECT_NAME@_LIBRARIES "")
+foreach(lib @_PKG_CONFIG_LIBS@)
+  # If the component is a link directory ("-L/full/path"), skip.
+  string(FIND ${lib} "-L" _is_library_dir)
+  if(${_is_library_dir} EQUAL -1)
+    string(REPLACE "-l" "" lib ${lib})
+    find_library(abs_lib_${lib} ${lib} HINTS ${eigenpy_LIBRARY_DIRS})
```
Comment (jcarpent) [inline Config.cmake.in]: I think `${eigenpy_LIBRARY_DIRS}` is not the correct HINT for a generic implementation.

Comment (wxmerkt) [reply Config.cmake.in]: Ah, thanks for noticing! That was a left-over from trying tons of different approaches... I'll get it fixed

**Comment by wxmerkt** [comment]
> @wxmerkt I think the procedure you introduced in Config.cmake.in should rather go to package-config.cmake in such a way you only export @PROJECT_NAME@_LIBRARIES. Could you apply this change?

Where do you propose this to be added in `package-config.cmake.in`?

**Comment by jcarpent** [comment]
I was suggesting to populate an extra variable in SETUP_PROJECT_PACKAGE_FINALIZE located in package-config.cmake

**Comment by wxmerkt** [comment]
@jcarpent Done :-)

**Pull request merged**